### PR TITLE
Add basic XR navigation ESM script

### PR DESCRIPTION
**Pull request by willeastcott**
One handed navigation in XR. Works with both hands and tracked controllers.

https://github.com/user-attachments/assets/0fbbe46c-132d-4c07-bd41-1584d6faaf99

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: scripts/esm/xr-navigation.mjs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            0,
+            origin.z + direction.z * t
+        );
+    }
+
+    tryTeleport(inputSource) {
+        const origin = inputSource.getOrigin();
+        const direction = inputSource.getDirection();
+
+        const hitPoint = this.findPlaneIntersection(origin, direction);
```
Comment (mvaligursky) [inline scripts/esm/xr-navigation.mjs:70]: personally I'd just use `Plane.intersectsRay` here, but this works too (and is faster)

File: scripts/esm/xr-navigation.mjs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            if (!this.activePointers.get(inputSource)) continue;
+
+            const start = inputSource.getOrigin();
+            const direction = inputSource.getDirection();
+
+            const hitPoint = this.findPlaneIntersection(start, direction);
+
+            if (hitPoint) {
+                // Draw line to intersection point
+                this.app.drawLine(start, hitPoint, this.validColor);
```
Comment (mvaligursky) [inline scripts/esm/xr-navigation.mjs:90]: you could move all the drawing code to a separate function, allowing users to easily extend the script and implement a custom drawing by overriding the function

**Review by mvaligursky — APPROVED** [review]

**Review by slimbuck — APPROVED** [review]
<3<3

File: scripts/esm/xr-navigation.mjs [hunk 3]
```diff
@@ -0,0 +1,123 @@
+/* eslint-disable-next-line import/no-unresolved */
+import { Color, Script, Vec3 } from 'playcanvas';
+
+/** @import { XrInputSource } from 'playcanvas' */
+
+export class XrNavigation extends Script {
```
[the remark this answers is not in the document]
Comment (willeastcott) [reply scripts/esm/xr-navigation.mjs:16]: I've added a class description.

File: scripts/esm/xr-navigation.mjs (hunk 2 above)
Comment (willeastcott) [reply scripts/esm/xr-navigation.mjs:100]: Yes, customizing visuals is important but I'd like to leave solving that to a follow up PR.

File: scripts/esm/xr-navigation.mjs (hunk 1 above)
Comment (willeastcott) [reply scripts/esm/xr-navigation.mjs:80]: I'll keep that in mind for a follow up PR.

**Pull request merged**